### PR TITLE
Github Actions run Jest unit and integration tests on every push

### PR DESCRIPTION
**Pull request by RebeccaStankus**

### CODE REVIEW

File: .github/workflows/run-unit-tests.yml [hunk 1]
```diff
@@ -1,37 +1,68 @@
...
       - main
+      - gha
+ 
 
 jobs:
-  Run-Jest-Unit-Tests:
+  Run-Jest-Tests:
     runs-on: ubuntu-latest
-    if: github.event.pusher.name != 'hifibuild'
+    if: github.event.pusher.namen != 'hifibuild'
```
Comment (luiscuenca) [inline .github/workflows/run-unit-tests.yml:22]: Is this a typo?

Comment (RebeccaStankus) [reply .github/workflows/run-unit-tests.yml:22]: oh, yeah...thanks!

**Review by luiscuenca — DISMISSED** [review]

**Review by luiscuenca — APPROVED** [review]

**Review by joy-highfidelity — APPROVED** [review]
✨

**Review by hd23408 — COMMENTED** [review]
This is exciting! Thank you @RebeccaStankus !

I'd like to have us NOT run the admin REST API tests on every push to hifi-spatial-audio-js, because there isn't anything that could be changed in the hifi-spatial-audio-js code that would affect the behavior of REST API calls.

So, I think that tests/integration/rest.integration.test.ts already separates out all the REST API tests, correct? If so, what I'd like to happen is for us NOT to run the REST tests as part of the GHA runs, and instead just run the HiFiCommunicator.integration.test.ts tests. Thanks!

**Comment by RebeccaStankus** [comment]
@hd23408 I went ahead and reintroduced 'smoke' tests for the category we run on deploy and put REST tests in there.

**Review by luiscuenca — APPROVED** [review]

**Pull request merged**